### PR TITLE
Mention restriction on usage of UnmangedType.CustomMarshaler

### PR DESCRIPTION
**Pull request by dlech**
Copy sentence from UnmangedType.LPStruct to UnmangedType.CustomMarshaler since the same restriction applies.

See dotnet/coreclr#12029

### CODE REVIEW

File: xml/System.Runtime.InteropServices/UnmanagedType.xml [hunk 1]
```diff
@@ -291,7 +291,7 @@
         <ReturnType>System.Runtime.InteropServices.UnmanagedType</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Specifies the custom marshaler class when used with the <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalType" /> or <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalTypeRef" /> field. The <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalCookie" /> field can be used to pass additional information to the custom marshaler. You can use this member on any reference type.</summary>
+        <summary>Specifies the custom marshaler class when used with the <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalType" /> or <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalTypeRef" /> field. The <see cref="F:System.Runtime.InteropServices.MarshalAsAttribute.MarshalCookie" /> field can be used to pass additional information to the custom marshaler. You can use this member on any reference type. This member is valid for platform invoke methods only.</summary>
```
Comment (yizhang82) [inline xml/System.Runtime.InteropServices/UnmanagedType.xml]: This also works on COM interop. So this statement is not accurate.
How about: 
This member is valid for parameters and return values only. It cannot be used on struct fields.

**Review by yizhang82 — CHANGES_REQUESTED** [review]

**Comment by mairaw** [comment]
@dlech let us know if you'll make the changes suggested by @yizhang82. Otherwise, we can make them for you in this PR. Thanks!

**Comment by dlech** [comment]
Whoops, forgot about this one. I have added the change now.

**Review by yizhang82 — APPROVED** [review]
LGTM. Thank you!

**Review by rpetrusha — APPROVED** [review]

**Pull request merged**